source/grid/grid_in.cc

```diff
@@ -340,139 +340,181 @@ GridIn<dim, spacedim>::read_vtk(std::istream &in)
-                              else if (set == "ManifoldID")
-                                boundary_line.manifold_id =
-                                  static_cast<types::manifold_id>(id);
-                              else
-                                Assert(false, ExcInternalError());
-                            }
-                        }
-                      else if (dim == 2)
-                        {
-                          for (auto &boundary_line : subcelldata.boundary_lines)
-                            {
-                              double id;
-                              in >> id;
-                              if (set == "MaterialID")
-                                boundary_line.material_id =
-                                  static_cast<types::material_id>(id);
-                              else if (set == "ManifoldID")
-                                boundary_line.manifold_id =
-                                  static_cast<types::manifold_id>(id);
-                              else
-                                Assert(false, ExcInternalError());
-                            }
-                        }
-                    }
+                double id;
+                in >> id;
+                if (set == "MaterialID")
+                  boundary_quad.material_id = static_cast<types::material_id>(id);
+                else if (set == "ManifoldID")
+                  boundary_quad.manifold_id = static_cast<types::manifold_id>(id);
+                else
+                  Assert(false, ExcInternalError());             
+              }
+              for (auto &boundary_line : subcelldata.boundary_lines)
+              {
+                double id;
+                in >> id;
+                if (set == "MaterialID")
+                  boundary_line.material_id = static_cast<types::material_id>(id);
+                else if (set == "ManifoldID")
+                  boundary_line.manifold_id = static_cast<types::manifold_id>(id);
+                else
+                  Assert(false, ExcInternalError());
+              }
+            }
+            else if (dim == 2)
+            {
+              for (auto &boundary_line : subcelldata.boundary_lines)
+              {
+                double id;
+                in >> id;
+                if (set == "MaterialID")
+                  boundary_line.material_id = static_cast<types::material_id>(id);
+                else if (set == "ManifoldID")
+                  boundary_line.manifold_id = static_cast<types::manifold_id>(id);
+                else
+                  Assert(false, ExcInternalError());             
               }
+            }
+            in >> keyword;
+        }
+      }
+      
+      //////////////////
+      //// Addition of FIELD DATA:
+      //////////////////      
+      if(keyword == "FIELD")
+      {
+        std::cout<<std::endl<<"Processing FIELD_DATA...."<<std::endl;
+        unsigned int no_fields;
```

Comment:
Our usual coding style is to name these variables `n_fields`, to ensure that it's not misread as "[There are] no fields", which would then have to be a `bool` variable.